bundles/org.openhab.binding.gardena/src/main/java/org/openhab/binding/gardena/internal/GardenaSmartWebSocket.java

```diff
@@ -139,18 +136,29 @@ public void onFrame(Frame pong) {
     @OnWebSocketClose
     public void onClose(int statusCode, String reason) {
         if (!closing) {
-            logger.debug("Connection to Gardena Webservice was closed ({}): code: {}, reason: {}", socketId, statusCode,
+            logger.warn("Connection to Gardena Webservice was closed ({}): code: {}, reason: {}", socketId, statusCode,
```

Comment:
I think your problem is that onClose() will be called by one of two ways. Either because a ping failed, or because the access token expired.

The latter is a normal occurrence, (it occurs every two hours), so it should not issue a logger.warn() message; it should probably only issue a logger.debug() message as it was in the prior code.

By contrast, perhaps a failed ping should produce a higher level of logger output; perhaps logger.info(). But even a failed ping might be considered as a normal occurrence, so I am not even sure if logger.info() would be justified.

I would suggest that really the only case where you should issue logger.warn() is when the connection was closed AND the subsequent re-connection attempt failed.

Reply:
I agree that a connection loss because of an expired token should not produce a `logger.warn()`. But I think any other disconnection is not a normal behavior and the user should be informed about it. In my openhab installation I sometimes see disconnections with the message `"code: 1006, reason: Disconnected"` every 2 minutes for multiple hours...

My suggestion would be: Use `logger.debug` for an expired token (`code: 1001, reason: Going away`) and `logger.info` for any other error.

Reply:
> Use logger.debug for an expired token (code: 1001, reason: Going away) and logger.info for any other error.

Ok. Makes sense.